scalecube-organization/src/main/java/io/scalecube/organization/repository/couchbase/CouchbaseRepository.java

```diff
@@ -1,183 +1,169 @@
-    }
-  }
-
-  protected Bucket client() {
-    return cluster.openBucket(bucketName, settings.password());
-  }
-
-  private <R> Observable<R> executeAsync(Observable<R> asyncAction) {
-    return asyncAction.onErrorResumeNext(
-        e -> {
-          if (e instanceof RuntimeException) {
-            return Observable.error(
-                exceptionTranslator.translateExceptionIfPossible((RuntimeException) e));
-          } else if (e instanceof TimeoutException) {
-            return Observable.error(new QueryTimeoutException(e.getMessage(), e));
-          } else if (e instanceof InterruptedException) {
-            return Observable.error(new OperationInterruptedException(e.getMessage(), e));
-          } else if (e instanceof ExecutionException) {
-            return Observable.error(new OperationInterruptedException(e.getMessage(), e));
-          } else {
-            return Observable.error(e);
-          }
-        });
-  }
-
-  protected <R> R execute(BucketCallback<R> action, Bucket client) {
-    requireNonNull(client);
-    requireNonNull(action);
-
-    try {
-      return action.doInBucket();
-    } catch (RuntimeException ex) {
-      throw exceptionTranslator.translateExceptionIfPossible(ex);
-    } catch (TimeoutException ex) {
-      throw new QueryTimeoutException(ex.getMessage(), ex);
-    } catch (InterruptedException | ExecutionException ex) {
-      throw new OperationInterruptedException(ex.getMessage(), ex);
-    } finally {
-      client.close();
-    }
-  }
-}
+package io.scalecube.organization.repository.couchbase;
+
+import static com.couchbase.client.java.query.Select.select;
+import static com.couchbase.client.java.query.dsl.Expression.i;
+import static java.util.Objects.requireNonNull;
+
+import com.couchbase.client.java.Bucket;
+import com.couchbase.client.java.document.JsonDocument;
+import com.couchbase.client.java.document.RawJsonDocument;
+import com.couchbase.client.java.query.N1qlQuery;
+import com.couchbase.client.java.query.N1qlQueryResult;
+import com.couchbase.client.java.query.N1qlQueryRow;
+import com.couchbase.client.java.query.SimpleN1qlQuery;
+import io.scalecube.organization.domain.Entity;
+import io.scalecube.organization.repository.Repository;
+import io.scalecube.organization.repository.exception.DataRetrievalFailureException;
+import io.scalecube.organization.repository.exception.OperationInterruptedException;
+import io.scalecube.organization.repository.exception.QueryTimeoutException;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeoutException;
+import rx.Observable;
+
+/**
+ * Abstract base couchbase <Code>Repository</Code> implementation.
+ *
+ * @param <T> This repository entity type.
+ * @param <I> This repository entity Id type which extends <codes>java.lang.String</codes>.
+ */
+abstract class CouchbaseRepository<T extends Entity, I extends String> implements Repository<T, I> {
```

Comment:
`String` is `final` in java. Can we just remove `I`?

Reply:
Done.